java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -50,6 +53,78 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_sequence(JNIEnv *env, j
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_fromArrow(JNIEnv *env, jclass,
+                                                                   jint j_type,
+                                                                   jlong j_col_length,
+                                                                   jlong j_null_count,
+                                                                   jobject j_data_obj,
+                                                                   jobject j_validity_obj,
+                                                                   jobject j_offsets_obj) {
+  try {
+    cudf::jni::auto_set_device(env);
+    cudf::type_id n_type = static_cast<cudf::type_id>(j_type);
+    // not all the buffers are used for all types
+    void const *data_address = 0;
+    int data_length = 0;
+    if (j_data_obj != 0) {
+      data_address = env->GetDirectBufferAddress(j_data_obj);
+      data_length = env->GetDirectBufferCapacity(j_data_obj);
+    }
+    void const *validity_address = 0;
+    int validity_length = 0;
+    if (j_validity_obj != 0) {
+      validity_address = env->GetDirectBufferAddress(j_validity_obj);
+      validity_length = env->GetDirectBufferCapacity(j_validity_obj);
+    }
+    void const *offsets_address = 0;
+    int offsets_length = 0;
+    if (j_offsets_obj != 0) {
+      offsets_address = env->GetDirectBufferAddress(j_offsets_obj);
+      offsets_length = env->GetDirectBufferCapacity(j_offsets_obj);
+    }
+    auto data_buffer = arrow::Buffer::Wrap(static_cast<const char *>(data_address), static_cast<int>(data_length));
+    auto null_buffer = arrow::Buffer::Wrap(static_cast<const char *>(validity_address), static_cast<int>(validity_length));
+    auto offsets_buffer = arrow::Buffer::Wrap(static_cast<const char *>(offsets_address), static_cast<int>(offsets_length));
+
+    cudf::jni::native_jlongArray outcol_handles(env, 1);
+    std::shared_ptr<arrow::Array> arrow_array;
+    switch (n_type) {
+      case cudf::type_id::DECIMAL32:
+        JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "Don't support converting DECIMAL32 yet", 0);
```

Comment:
Nit: to reduce the chance of typos on this class, there's a `cudf::jni::ILLEGAL_ARG_CLASS` definition in `jni_utils.hpp`.

Reply:
used